spyder/utils/qthelpers.py

```diff
@@ -502,7 +502,42 @@ def show_std_icons():
     dialog.show()
     sys.exit(app.exec_())
 
-    
+
+def create_control_layout(controls, main_widget):

```

Comment:
I've intentionally chosen the first approach. First, you would even need `create_control_layout(widgets=None, layout=None, main_widget=None)` and say: "provide `main_widget` and either `widgets` or `layout`. Alternatively, putting `main_widgets` first as positional argument is also not very nice, because it does not reflect the natural order in which the widgets will be drawn.

Secondly, I think it's perfectly right to use just one variable for "the things to put at the top", no matter what this actually is. Having the ability to put either a list of widgets or a layout is a suitable generalization.